cmd/cmd_volume.go

```diff
@@ -225,13 +227,130 @@ func volumeTreeCommandHandler(cmd *cobra.Command, args []string) {
 	reader.Close()
 }
 
+func volumeLsCommand() *cobra.Command {
+	var cmdLs = &cobra.Command{
+		Use:   "ls <volume_name:volume_uuid> <path>",
+		Short: "list files and directories in volume",
+		Run:   volumeLsCommandHandler,
+		Args:  cobra.MinimumNArgs(1),
+	}
+	flags := cmdLs.PersistentFlags()
+	flags.BoolP("long-format", "l", false, "use a long listing format")
+	return cmdLs
+}
+
+func volumeLsCommandHandler(cmd *cobra.Command, args []string) {
```

Comment:
This function is identical to imageLsCommandHandler() except for the call to getLocalVolumeReader() instead of getLocalImageReader(). I would make a more generic `imageLs(flags *pflag.FlagSet, args []string, reader *fs.Reader)` function and use it in both imageLsCommandHandler() and volumeLsCommandHandler().

Reply:
the plan was to refactor those a bit late for reuse, but will do it now, thanks